### PR TITLE
coordinator: checks all scanned regions in a patrol

### PR DESCRIPTION
**Pull request by nolouch**
checks all scanned regions in a patrol and adds a metrics.

### CODE REVIEW

File: server/coordinator.go [hunk 1]
```diff
@@ -143,12 +143,14 @@ func (c *coordinator) patrolRegions() {
 
 			key = region.GetEndKey()
 
-			if c.checkRegion(region) {
-				break
-			}
+			c.checkRegion(region)
```
Comment (siddontang) [inline server/coordinator.go]: can it cause other operations (not only for removing nodes) too fast?

**Comment by siddontang** [comment]
PTAL @huachaohuang @disksing

File: server/coordinator.go (hunk 1 above)
Comment (disksing) [reply server/coordinator.go]: I don't think it is a good fix. As we have schedule limit, when there are many regions need to update, only operators for the first serval regions can be created. The rest will not be handled util scan them the next time.

There are 2 approaches could be better:
1) decrease sleep time to 10ms and make it configurable.
2) don't sleep if an operator is successfully created.

File: server/coordinator.go [hunk 2]
```diff
@@ -141,14 +141,17 @@ func (c *coordinator) patrolRegions() {
 				continue
 			}
 
-			key = region.GetEndKey()
-
-			if c.checkRegion(region) {
+			if !c.checkRegion(region) {
```
Comment (disksing) [inline server/coordinator.go]: When it returns false, it does not mean the created operator is not successfully added. It's also possible that no operator is needed.

Comment (huachaohuang) [reply server/coordinator.go]: Why do we break here? What about the other scanned regions?

File: server/coordinator.go [hunk 3]
```diff
@@ -141,20 +141,25 @@ func (c *coordinator) patrolRegions() {
...
+			start = time.Now()
+		}
 	}
 }
 
+// checkRegion returns false if region produced an operator and fail to add the operator.
 func (c *coordinator) checkRegion(region *core.RegionInfo) bool {
 	// If PD has restarted, it need to check learners added before and promote them.
 	// Don't check isRaftLearnerEnabled cause it may be disable learner feature but still some learners to promote.
+	var op, op1, op2 *schedule.Operator
```
Comment (disksing) [inline server/coordinator.go]: op1 and op2 will not set.

**Comment by nolouch** [comment]
I wii use configuable interval  in another pr.

**Pull request closed without merging**